### ISSUE TITLE
Test errors with 'too many open files'

### ISSUE THREAD

**Issue by Duologic** [open-policy-agent/conftest#427]
Awesome tool, should replace a bunch of my grep/awk/bash-fu scripts.

I just ran it against a set of ~4000 files and received this error:
```
Error: running test: get configurations: get config: open file: open /redacted/path/to/file.yaml: too many open files
```

Had a quick look at the code, it seems to close the files early (not deferring) but maybe there is some parallelization going on? Any hints on or help on this, wouldn't mind coming up with a PR to fix it.

Installed on Arch Linux through the AUR pacakge:
```
# conftest --version
Version: 0.21.0
Commit:
Date:
```

**Comment by jpreese** [comment]
@Duologic could you try this scenario with the unreleased version of Conftest? 

There have been a lot of optimizations since `0.21.0` and this may already be resolved. I am seeing a lot of .Open calls in `0.21.0`.

**Comment by sudermanjr** [comment]
I was running into this issue as well, and building off master really helped.

**Comment by carlossg** [comment]
FWIW This is fixed in 0.23.0

**Comment by artis3n** [comment]
Can confirm that moving from 0.21.0 to 0.23.0 allows 5067 yaml files to be scanned without issue :)
Well...it takes like a minute but no more errors.
